version: '3.7'
services:
  portainer:
    image: portainer/portainer-ee:latest
    container_name: portainer
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    networks:
       - DMZ
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ${DOCKER_VOLUME_PATH}/portainer:/data
    ports:
      - 9443:9443 
      - 8000:8000
    environment:
      - TZ=${TZ}
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.portainer-rtr.rule=Host(`portainer.${SUB_DOMAIN_1}`)"
      - "traefik.http.routers.portainer-rtr.entrypoints=https"
      - "traefik.http.services.portainer-rtr.loadbalancer.server.port=9443"
      - "traefik.http.services.portainer-rtr.tls={}"
      - "traefik.http.routers.portainer-rtr.tls=true"
      - "traefik.http.routers.portainer-rtr.tls.certResolver=azure"
      - "traefik.http.routers.portainer-rtr.tls.domains[0].main=${DOMAIN_1}"
      - "traefik.http.routers.portainer-rtr.tls.domains[0].sans=*.${SUB_DOMAIN_1}"
      - "traefik.docker.network=DMZ"

networks:
  DMZ:
    external:
      name: DMZ